### PR TITLE
mgr/dashboard: Improve level A accessibility for grafana component

### PR DESCRIPTION
**Pull request by nSedrickm**
*Add descriptive iframe titles for embedded grafana panels

Fixes: https://tracker.ceph.com/issues/55890
Signed-off-by: nsedrickm <nsedrick101@gmail.com>

image

## Contribution Guidelines
- To sign and title your commits, please refer to [Submitting Patches to Ceph](https://github.com/ceph/ceph/blob/main/SubmittingPatches.rst).

- If you are submitting a fix for a stable branch (e.g. "pacific"), please refer to [Submitting Patches to Ceph - Backports](https://github.com/ceph/ceph/blob/master/SubmittingPatches-backports.rst) for the proper workflow.

## Checklist
- Tracker (select at least one)
  - [x] References tracker ticket
  - [ ] Very recent bug; references commit where it was introduced
  - [ ] New feature (ticket optional)
  - [ ] Doc update (no ticket needed)
  - [ ] Code cleanup (no ticket needed)
- Component impact
  - [x] Affects [Dashboard](https://tracker.ceph.com/projects/dashboard/issues/new), opened tracker ticket
  - [ ] Affects [Orchestrator](https://tracker.ceph.com/projects/orchestrator/issues/new), opened tracker ticket
  - [ ] No impact that needs to be tracked
- Documentation (select at least one)
  - [ ] Updates relevant documentation
  - [x] No doc update is appropriate
- Tests (select at least one)
  - [x] Includes [unit test(s)](https://docs.ceph.com/en/latest/dev/developer_guide/tests-unit-tests/)
  - [ ] Includes [integration test(s)](https://docs.ceph.com/en/latest/dev/developer_guide/testing_integration_tests/)
  - [ ] Includes bug reproducer
  - [ ] No tests

<details>
<summary>Show available Jenkins commands</summary>

- `jenkins retest this please`
- `jenkins test classic perf`
- `jenkins test crimson perf`
- `jenkins test signed`
- `jenkins test make check`
- `jenkins test make check arm64`
- `jenkins test submodules`
- `jenkins test dashboard`
- `jenkins test dashboard cephadm`
- `jenkins test api`
- `jenkins test docs`
- `jenkins render docs`
- `jenkins test ceph-volume all`
- `jenkins test ceph-volume tox`
- `jenkins test windows`
</details>

### CODE REVIEW

**Review by ljflores — APPROVED** [review]
Looks good; agree with @Pegonzal about adding `i18n-title`.

**Review by epuertat — APPROVED** [review]
LGTM! Just pending the i18n on the title field. Any chance to cover this with e2e or linter tests_

**Comment by nSedrickm** [comment]
@epuertat the search for an accessibility linter is a bit challenging for our usecase. I haven't really found a good match. Would have been great if the [ axe vscode plugin](https://marketplace.visualstudio.com/items?itemName=deque-systems.vscode-axe-linter) came with a cli :)

We could start by enabling [codelyzer a11y](https://web.dev/accessible-angular-with-codelyzer/#using-codelyzer-to-detect-inaccessible-elements) rule(s) which will also be maintained when we do move to [eslint](https://github.com/angular-eslint/angular-eslint#functionality). Also found this [tool](https://github.com/musienkoyuriy/enabler/) works too but not sure about its support

I am currently running down these lists for research
https://www.w3.org/WAI/ER/tools/
https://github.com/brunopulis/awesome-a11y/blob/main/topics/tools.md

File: src/pybind/mgr/dashboard/frontend/src/app/ceph/cephfs/cephfs-tabs/cephfs-tabs.component.html [hunk 1]
```diff
@@ -35,7 +35,8 @@
       <a ngbNavLink
          i18n>Performance Details</a>
       <ng-template ngbNavContent>
-        <cd-grafana [grafanaPath]="'mds-performance?var-mds_servers=mds.' + grafanaId"
+        <cd-grafana title="CephFS mds performance"
```
[the remark this answers is not in the document]
Comment (nSedrickm) [reply src/pybind/mgr/dashboard/frontend/src/app/ceph/cephfs/cephfs-tabs/cephfs-tabs.component.html]: Updated :)

**Comment by nSedrickm** [comment]
> Hey @nSedrickm!
> 
> All looks good to me. But I'm wondering, shouldn't we include `i18n-title` property so the title can be translated?

@ljflores @Pegonzal, Updated :)

**Comment by ljflores** [comment]
jenkins test make check

**Comment by nSedrickm** [comment]
> @nSedrickm great! Last thing to address is Ernesto's comment about covering this with e2e or linter tests.

Working on this, left a comment above regarding the linters

**Comment by ljflores** [comment]
> Working on this, left a comment above regarding the linters

Awesome!

**Comment by epuertat** [comment]
> @epuertat the search for an accessibility linter is a bit challenging for our usecase. I haven't really found a good match. Would have been great if the [ axe vscode plugin](https://marketplace.visualstudio.com/items?itemName=deque-systems.vscode-axe-linter) came with a cli :)
> 

It's surprising that they provide a vscode plugin but not the core code that's running upon. Perhaps they sell that tool? Would you mind sending them an email? It's also a good task for a GSoC internship to get some exposure to vendors and commercial solutions too. For example, for last year's GSoC Applitools provided us with an OpenSource access to their tool. 

> We could start by enabling [codelyzer a11y](https://web.dev/accessible-angular-with-codelyzer/#using-codelyzer-to-detect-inaccessible-elements) rule(s) which will also be maintained when we do move to [eslint](https://github.com/angular-eslint/angular-eslint#functionality). Also found this [tool](https://github.com/musienkoyuriy/enabler/) works too but not sure about its support

Let's start with the codelizer rule since we already have tslint running in our CI. And the other ones looks interesting too.

> I am currently running down these lists for research [w3.org/WAI/ER/tools](https://www.w3.org/WAI/ER/tools/) [https://github.com/brunopulis/awesome-a11y/blob/main/topics/tools.md](https://github.com/brunopulis/awesome-a11y/blob/main/topics/tools.md?rgh-link-date=2022-07-20T20%3A08%3A20Z)

Nice resources! You're doing a really thorough job here @nSedrickm !

**Comment by nSedrickm** [comment]
> > @epuertat the search for an accessibility linter is a bit challenging for our usecase. I haven't really found a good match. Would have been great if the [ axe vscode plugin](https://marketplace.visualstudio.com/items?itemName=deque-systems.vscode-axe-linter) came with a cli :)
> 
> It's surprising that they provide a vscode plugin but not the core code that's running upon. Perhaps they sell that tool? Would you mind sending them an email? It's also a good task for a GSoC internship to get some exposure to vendors and commercial solutions too. For example, for last year's GSoC Applitools provided us with an OpenSource access to their tool.
> 
> > We could start by enabling [codelyzer a11y](https://web.dev/accessible-angular-with-codelyzer/#using-codelyzer-to-detect-inaccessible-elements) rule(s) which will also be maintained when we do move to [eslint](https://github.com/angular-eslint/angular-eslint#functionality). Also found this [tool](https://github.com/musienkoyuriy/enabler/) works too but not sure about its support
> 
> Let's start with the codelizer rule since we already have tslint running in our CI. And the other ones looks interesting too.
> 
> > I am currently running down these lists for research [w3.org/WAI/ER/tools](https://www.w3.org/WAI/ER/tools/) [https://github.com/brunopulis/awesome-a11y/blob/main/topics/tools.md](https://github.com/brunopulis/awesome-a11y/blob/main/topics/tools.md?rgh-link-date=2022-07-20T20%3A08%3A20Z)
> 
> Nice resources! You're doing a really thorough job here @nSedrickm !

Thanks, didn't even think of that. Will send them one right away!

**Comment by epuertat** [comment]
jenkins retest this please

**Comment by nSedrickm** [comment]
jenkins retest this please

**Comment by ljflores** [comment]
jenkins retest this please

**Comment by epuertat** [comment]
jenkints test api

File: src/pybind/mgr/dashboard/frontend/tslint.json [hunk 2]
```diff
@@ -112,7 +113,12 @@
           }
         ]
       }
-    ]
+    ],
+    "template-accessibility-alt-text": true,
+    "template-accessibility-tabindex-no-positive": true,
+    "template-accessibility-table-scope": true,
+    "template-accessibility-valid-aria": true,
+    "template-no-distracting-elements": true
```
Comment (epuertat) [inline src/pybind/mgr/dashboard/frontend/tslint.json:121]: Nice! Are you aware of any well-know `tslint.conf` (like [Microsoft's tslint.conf](https://github.com/microsoft/tslint-microsoft-contrib)) where these are enabled by default?

Comment (nSedrickm) [reply src/pybind/mgr/dashboard/frontend/tslint.json:121]: Unfortunately no, these rules are opt in only. One repo where they are mostly mentioned is [ tslint-to-eslint-config](https://github.com/typescript-eslint/tslint-to-eslint-config)

**Comment by ljflores** [comment]
jenkins retest this please

**Comment by nSedrickm** [comment]
jenkins test make check

**Comment by nSedrickm** [comment]
jenkins test api

**Comment by nSedrickm** [comment]
jenkins test make check

**Comment by nSedrickm** [comment]
jenkins test make check

**Comment by epuertat** [comment]
jenkins test make check

**Comment by nSedrickm** [comment]
jenkins test make check

**Comment by nSedrickm** [comment]
jenkins test dashboard cephadm

**Comment by nSedrickm** [comment]
jenkins test make check arm64

**Comment by nSedrickm** [comment]
jenkins test windows

**Comment by nSedrickm** [comment]
jenkins test make check

**Comment by nSedrickm** [comment]
jenkins test make check

**Comment by nSedrickm** [comment]
jenkins test make check arm64

**Comment by nSedrickm** [comment]
jenkins test make check

**Comment by nSedrickm** [comment]
jenkins test make check

**Comment by ljflores** [comment]
jenkins test make check

**Comment by nSedrickm** [comment]
Hi @pereman2 this one is good to go!

**Pull request merged**